### PR TITLE
Updated entities with some fields and with update mixin

### PR DESCRIPTION
**Pull request by sghai**
Updated 3 entities:
- `Environment` with update mixin
- `media` with update mixin
- `SmartProxy` with update mixin and `location` and `organization` fields

### CODE REVIEW

**Comment by sghai** [comment]
`Media test results`:

```
In [5]: location = entities.Location(server_config, name="gh3").create()
In [6]: media = entities.Media(server_config).search(query={'organization_id': '21'})[0]
In [7]: media.location = [location]
In [8]: media = media.update(['location'])

In [10]: media.location
Out[10]: [nailgun.entities.Location(nailgun.config.ServerConfig(url='https://foreman_server', verify=False, auth=('admin', 'changeme')), id=28)]
```

**Comment by sghai** [comment]
`Environment test results`:

```
In [5]: location = entities.Location(server_config, name="gh6").create()
In [6]: environment = entities.Environment(server_config).search(query={'organization_id': '21'})[0]
In [7]: environment.location = [location]
In [8]: environment = environment.update(['location'])

In [10]: environment.location
Out[10]: [nailgun.entities.Location(nailgun.config.ServerConfig(url='https://foreman_server', verify=False, auth=('admin', 'changeme')), id=31)]
```

**Comment by sghai** [comment]
`SmartProxy test results`:

```
In [6]: location = entities.Location(server_config, name="gh9").create()
In [7]: proxy = entities.SmartProxy(server_config).search(query={u'search': u'name=foreman_server'})[0]
In [8]: proxy.location = [location]
In [9]: proxy = proxy.update(['location'])

In [10]: proxy.location
Out[10]: [nailgun.entities.Location(nailgun.config.ServerConfig(url='https://foreman_server', verify=False, auth=('admin', 'changeme')), id=34)]
```

**Comment by sghai** [comment]
Please note that I'm using `update()` method explicitly for all 3 entities since `PUT` request doesn't show all the attributes.

Filed github issues:
https://github.com/SatelliteQE/nailgun/issues/202
https://github.com/SatelliteQE/nailgun/issues/203

**Comment by sthirugn** [comment]
ACK

File: nailgun/entities.py [hunk 1]
```diff
@@ -2334,6 +2355,23 @@ def read(self, entity=None, attrs=None, ignore=None):
...
+    def update(self, fields=None):
+        """Fetch a complete set of attributes for this entity.
+
+        Beware of `Bugzilla #1236008
+        <https://bugzilla.redhat.com/show_bug.cgi?id=1236008>`_:
+        "Cannot use HTTP PUT to associate location with media"
+
+        """
+        self.update_json(fields)
+        return self.read()
```
Comment (Ichimonji10) [inline nailgun/entities.py]: [BZ 1236008](https://bugzilla.redhat.com/show_bug.cgi?id=1236008) describes how calling `Location.update` can trigger an HTTP 500. But this it the `Media.update` method. It this method really necessary? And if so, is there a more appropriate bug?

**Comment by Ichimonji10** [comment]
ACK pending comment.

File: nailgun/entities.py (hunk 1 above)
Comment (sghai) [inline nailgun/entities.py]: Yeah.. I was trying to associate Location with `Media` entity and got this:

```
---------------------------------------------------------------------------
MissingValueError                         Traceback (most recent call last)
<ipython-input-9-7423c818e549> in <module>()
----> 1 media = media.update(['location'])

MissingValueError: Cannot find a value for the "location" field. Searched for keys named ('location_ids', 'location', 'locations'), but available keys are [u'name', u'config_path', u'created_at', u'updated_at', 'path_', u'media_path', u'os_family', u'id', u'image_path'].
```

Comment (sghai) [inline nailgun/entities.py]: Thanks for clarifying on IRC.. that's a different issue. I'll file a new bz and will update the PR

**Comment by sthirugn** [comment]
ACK

**Comment by sghai** [comment]
I was playing with git reset and unknowingly ran that cmd on this branch and that closes the PR. I'll raise another PR.

**Pull request closed without merging**